README.md

```diff
@@ -1,80 +1,213 @@
+    py -3 -m venv .venv
+    .venv\scripts\activate
+    ```
 
-To achieve this, when users submit a message to the web server, the web server will create an agent, thread, and stream back a reply. The response contains `agent_id` and `thread_id` in cookies. As a result, each subsequent message sent to the web server will also contain these IDs. As long as the same agent is being used in the system and the thread can be retrieved in the cookie, the same thread will be used to serve the users.
+    On Linux:
 
-## Local Development
+    ```shell
+    python3 -m venv .venv
+    source .venv/bin/activate
+    ```
 
-1. Run `pip install -r ./src/requirements.txt`.
+2. Navigate to the `src` directory:
 
-2. Make sure that the `.env` file exists.
+    ```shell
+    cd src
+    ```
 
-3. Store the Azure AI Projects connection string in the `.env` file as `PROJECT_CONNECTION_STRING`.
+3. Install required Python packages:
 
-4. Run `az login`.
+    ```shell
+    python -m pip install -r requirements.txt
+    ```
 
-5. Start the services with this command:
+4. Run the local server:
 
     ```shell
-    python -m uvicorn "api.main:create_app" --app-dir src --port 50505 --reload
+    python -m uvicorn "api.main:create_app" --factory --reload
     ```
 
-6. Click 'http://localhost:50505' in the browser to run the application.
+5. Click 'http://127.0.0.1:8000' in the terminal, which should open a new tab in the browser.
+
+6. Enter your message in the box.
+
+## Guidance
+
+### Troubleshooting
+
+#### Provisioning and Deployment Failures
+* If you have an issue is with timeouts or provisioning resources, changing the location of your resource group can help, as there may be availability constrains for resources. Call `azd down` and remove your current resources, and delete the `.azure` folder from your workspace. Then, call `azd up` again and select a different region. 
+* You may debug further using [azd commands](https://learn.microsoft.com/azure/developer/azure-developer-cli/reference#azd-deploy). `azd show` displays information abour your app and resources, and `azd deploy --debug` enables debugging and logging while deploying the application's code to Azure. 
+* Ensure that your az and azd tools are up to date.
+* After fully deploying with azd, additional errors in the Azure Portal may indicate that your latest code has not been successfully deployed
+
+#### Azure Container Apps
+* If your ACA does not boot up, it is possible that your deployment has failed. This could be due to quota constraints, permission issues, or resource availability. Check failures in the deployment and container app logs in the Azure Portal.
+* Console traces in ACA can be found in the Azure Portal, but they may be unreliable. Use Python’s logging with INFO level, and adjust Azure HTTP logging to WARNING.
+* Once your ACA is deployed, utilize the browser debugger (F12) and clear cache (CTRL+SHIFT+R). This can help debug the frontend for better traceability.
+
+#### Agents
```

Comment:
Thanks for this write up, let's check this again once the document is aligned with the other template.